lib/grpc/client/adapters/mint/connection_process/connection_process.ex

```diff
@@ -0,0 +1,177 @@
+defmodule GRPC.Client.Adapters.Mint.ConnectionProcess do
+  use GenServer
+
+  alias GRPC.Client.Adapters.Mint.ConnectionProcess.State
+  alias GRPC.Client.Adapters.Mint.StreamResponseProcess
+
+  require Logger
+
+  def start_link(scheme, host, port, opts \\ []) do
+    GenServer.start_link(__MODULE__, {scheme, host, port, opts})
+  end
+
+  def request(pid, method, path, headers, body, opts \\ []) do
+    GenServer.call(pid, {:request, method, path, headers, body, opts})
+  end
+
+  def disconnect(pid) do
+    GenServer.call(pid, {:disconnect, :brutal})
```

Comment:
why `:brutal`?

Reply:
Ideally in the future we could add logic here to wait for pending request to finish before killing the connection. `brutal` here refers to the fact that I'm going to kill the connection without wait for pending requests. thoughts?

Reply:
Perhaps just use `:disconnect` for now. Any refactors in the future can think of separate terminology depending on the use case.

Reply:
brutal/friendly doesn't seem like the right terminology for this

Reply:
Sure! I'll do that update